src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexMatcher.cs

```diff
@@ -233,6 +290,36 @@ private SymbolicRegexMatcher(SymbolicRegexBuilder<TSet> builder, SymbolicRegexNo
             }
             _reverseInitialStates = reverseInitialStates;
 
+
+            // Some SearchValues<char> implementations are slower than a DFA,
+            // but depend on input frequency.
+            // This is currently tuned for consistency
+            // but it could return false to enable findOptimizations.
+            bool CharSetIsTooCommon(RegexFindOptimizations.FixedDistanceSet fixedDistanceSet)
+            {
+                char[]? chars = fixedDistanceSet.Chars;
+                bool avoidSearchValues = false;
+                if (chars is not null && chars.Length > 5)
+                {
+                    // RegexFindOptimizations picks 3 sets at most so the construction overhead should not be too high
+                    var searchValues = SearchValues.Create(chars);
+                    avoidSearchValues = searchValues.GetType().Name switch
```

Comment:
I defer to @stephentoub but switching on the concrete type is fragile. 

Does this suggest there should be something exposed on a SearchValues object indicating what it is optimized for? (That would have to be API reviewed and come later)

Given that does not exist, should we debug.assert in the 'false' case that the type name is the case(s) expected today, so that if the internal type names like `ProbabilisticCharSearchValues` are renamed, this doesn't quietly fall down the slow path.

Same for ``GetType().Name != "StringSearchValuesAhoCorasick`2"``

Reply:
Since these are all fallback cases, i think the only `SearchValues<char>` implementation over 5 chars which should return false is ``RangeCharSearchValues`1``. Since not all of them are equal it would be good to know what the concrete implementation is.

If there were implementations for 2 ranges, or 1 range 1 char, then these should definitely be enabled for the engine as well.

We could also just disable above 5 chars since the range implementation ends up in the `{ Range: not null }` case, but i haven't confirmed this.

Reply:
I don't have an opinion on the approach simply that if it depends on internal names it should assert against the names changing.

Reply:
Ok, perhaps it's best to decide for non-bactracking inside the RegexFindOptimizations implementation